Providers/Profile.swift

```diff
@@ -985,9 +1073,12 @@ public class BrowserProfile: Profile {
 
                     return self.syncWith(remaining) >>== { deferMaybe(statuses + $0) }
                 }
-                reducer.terminal >>> self.endSyncing
 
-                // The actual work of synchronizing doesn't start until we append 
+                reducer.terminal.uponQueue(dispatch_get_main_queue()) { _ in
```

Comment:
Probably don't need to do this on the main queue and just use `upon` instead of `uponQueue`.